src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelBindingContextDisposable.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// Return value of <see cref="IModelBindingContext.PushContext"/>. Should be disposed
+    /// by caller when child binding context state should be popped off of 
+    /// the <see cref="IModelBindingContext"/>.
+    /// </summary>
+    public struct ModelBindingContextDisposable : IDisposable
```

Comment:
There are several places where methods are much more maintainable with a using statement. Keeping the context at the right state in the face of an uncaught exception is one case, another is when a method has non-trivial if/else or return flow control. The using statement is always implicitly correct.